src/main/java/net/kaczmarzyk/spring/data/jpa/domain/Join.java

```diff
@@ -51,10 +46,41 @@ public Join(QueryContext queryContext, String pathToJoinOn, String alias, JoinTy
 	@Override
 	public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query, CriteriaBuilder builder) {
 		query.distinct(distinctQuery);
-		queryContext.putLazyVal(alias, (r) -> r.join(pathToJoinOn, joinType));
+
+		if(!pathToJoinContainsAlias()) {
+			queryContext.putLazyVal(alias, (r) -> r.join(pathToJoinOn, joinType));
+		} else {
+			String[] pathToJoinOnSplittedByDot = pathToJoinOn.split("\\.");
+
+			if(pathToJoinOnSplittedByDot.length != 2) {
+				throw new IllegalArgumentException(
+						"Expected pathToJoin with single alias in pattern: 'alias.attribute' (without an apostrophe) where: " +
+						"alias is a alias of another join (of which annotation should be before annotation of actual join)," +
+						"attribute - name of the attribute for the target of the join."
+				);
+			}
+
+			String extractedAlias = pathToJoinOnSplittedByDot[0];
+			Object evaluated = queryContext.getEvaluated(extractedAlias, root);
+
+			if(evaluated instanceof javax.persistence.criteria.Join<?, ?>) {
+				String extractedPathToJoin = pathToJoinOnSplittedByDot[1];
+
+				queryContext.putLazyVal(
+						alias,
+						(r) -> ((javax.persistence.criteria.Join<?, ?>) evaluated).join(extractedPathToJoin, joinType)
+				);
+			} else {
+				queryContext.putLazyVal(alias, (r) -> r.join(pathToJoinOn, joinType));
```

Comment:
shouldn't we use "extractedPathToJoin" instead of "pathToJoinOn"? If not, then why?